pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerIdGenerator.java

```diff
@@ -287,4 +288,16 @@ private static String createLedgerPrefix(String ledgersPath, String idGenZnodeNa
         return ledgerIdGenPath + "/" + "ID-";
     }
 
+    //If the config rootPath when use zk metadata store, it will append rootPath as the prefix of the path.
+    //So when we get the path from the stat, we should truncate the rootPath.
+    private String handleTheDeletePath(String path) {
+        if (store instanceof ZKMetadataStore) {
+            String rootPath = ((ZKMetadataStore) store).getRootPath();
```

Comment:
I see we already have `ledgersRoot` of `PulsarLedgerIdGenerator` constructor, it's better to use it directly to avoid such checks here.

Reply:
And we don't need to introduce a new method in `ZKMetadataStore.java`

Reply:
> I see we already have ledgersRoot of PulsarLedgerIdGenerator constructor, it's better to use it directly to avoid such checks here.

It's not the zk rootPath, the `ledgersRoot` is `/ledegrs`.

Reply:
The `metadataServiceUriStr` is `metadata-store:127.0.0.1:2181/test`, it's not a standard schema, so we can't resolve the path correctly in `AbstractMetadataDriver#resolveLedgersRootPath`.

Reply:
I see.